test/e2e/parallel/web_console_plugin_test.go

```diff
@@ -0,0 +1,122 @@
+package parallel
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	v1 "github.com/openshift/api/route/v1"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func TestWebConsoleDeployedSuccessfully(t *testing.T) {
+	// given
+	await := WaitForDeployments(t)
+
+	for i, memberAwait := range await.AllMembers() {
+
+		signupRequest := NewSignupRequest(await).
+			Username(fmt.Sprintf("consoletest%d", i)).
+			Email("consoletest@redhat.com").
+			TargetCluster(memberAwait).
+			ManuallyApprove().
+			EnsureMUR().
+			RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+			Execute(t)
+		response := signupRequest.GetSignupResponse(t)
+
+		// Response should contain a ConsoleURL with a value something like:
+		// https://console-openshift-console.apps.99b682869228f7464338-mgmt.ci.hypershift.devcluster.openshift.com/
+		require.Contains(t, response, "consoleURL")
+
+		// Since we can't easily access the web console API resources directly (due to complex security requirements) we
+		// will instead create a route in the member cluster with which to access the console plugin
+
+		// Before we can create the route however, we need the certificate and private key values from the
+		// member-operator-console-plugin secret
+		secret, err := memberAwait.WaitForSecret(t, "member-operator-console-plugin")
+		require.NoError(t, err)
+
+		// Now we can create the route resource
+		route := &v1.Route{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "consolepluginroute",
+				Namespace: memberAwait.Namespace,
+				Annotations: map[string]string{
+					"openshift.io/host.generated": "true",
+				},
+			},
+			Spec: v1.RouteSpec{
+				To: v1.RouteTargetReference{
+					Kind: "Service",
+					Name: "member-operator-console-plugin",
```

Comment:
Should we wait for the `member-operator-console-plugin` service to exist before trying to create the route? Would it make the test more stable?

Reply:
and the paired PR: https://github.com/codeready-toolchain/member-operator/pull/431